JWTDecode/ValidatorJWT.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ValidatorJWT.swift
+//  JWTDecode
+//
+//  Copyright © 2018 Auth0. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+protocol ValidatorJWT {
+    var issuer: String { get }
+    var audience: String { get }
```

Comment:
Sounds like this `ValidatorJWT` should be a generic class, accepting **any claim** and not being limited to issuer / audience (as per the vars above) or nonce (as per the constructor below). The implementation `IDTokenValidation` however can be more specific. What were you trying to achieve? is this protocol definition correct?

Reply:
Why should it be generic? We can be opinionated.  Wouldn't we always expect to have an iss/aud?

Nonce is optional in this case below